app/models/solidus_stripe/gateway.rb

```diff
@@ -163,18 +165,20 @@ def request(&block)
 
     private
 
-    def create_confirmed_stripe_payment_intent(source, order, stripe_payment_intent_options:)
+    def create_confirmed_stripe_payment_intent(source, payment, stripe_payment_intent_options:)
       stripe_payment_method = source.stripe_payment_method
 
       SolidusStripe::PaymentIntent.create_stripe_intent(
         payment_method: source.payment_method,
-        order: order,
+        order: payment.order,
         stripe_intent_options: stripe_payment_intent_options.merge(
           payment_method: stripe_payment_method.id,
           customer: stripe_payment_method.customer,
           confirm: false
         )
-      ).tap { confirm_stripe_payment_intent(_1.id) }
+      )
+        .tap { confirm_stripe_payment_intent(_1.id) }
+        .tap { payment.update_column(:response_code, _1.id) } # rubocop:disable Rails/SkipsModelValidations
```

Comment:
Did you consider updating the response code before calling `confirm_stripe_payment_intent `?

Reply:
Oh, yeah, that was the whole point of this PR :sweat_smile: Changed, thanks for pointing out.